lib/generator/code_generator.dart

```diff
@@ -10,83 +10,112 @@ import 'package:graphql_generator/generator/objects.dart';
+        getTypesFromList(responseTypes, Kind.INPUT_OBJECT);
+    ObjectType mutationObjectType =
+        getMutationObjectType(responseTypes, mutationClassName);
+
+    return generateTypes(
+        namespace,
+        fragments,
+        types,
+        enumObjectTypes,
+        interfaceObjectTypes,
+        unionObjectTypes,
+        objectObjectTypes,
+        inputObjectTypes,
+        mutationObjectType);
+  }
+
+  List<ObjectType> getTypesFromList(
+      List<ObjectType> responseTypes, Kind typeKind) {
+    List<ObjectType> typeList =
+        responseTypes.where((type) => type.kind == typeKind).toList();
+    return typeList;
   }
 
-  splitTypes() {
-    enumTypes = responseTypes.where((type) => type.kind == Kind.ENUM).toList();
-    interfaceTypes =
-        responseTypes.where((type) => type.kind == Kind.INTERFACE).toList();
-    unionTypes =
-        responseTypes.where((type) => type.kind == Kind.UNION).toList();
-    inputObjectTypes =
-        responseTypes.where((type) => type.kind == Kind.INPUT_OBJECT).toList();
-    objectTypes =
-        responseTypes.where((type) => type.kind == Kind.OBJECT).toList();
-    mutation = responseTypes
+  ObjectType getMutationObjectType(
+      List<ObjectType> responseTypes, String mutationClassName) {
+    ObjectType mutationObjectType = responseTypes
         .where((type) => type.name == mutationClassName)
         .toList()
         .first;
+    return mutationObjectType;
   }
 
-  generateTypes() {
+  generateTypes(
+      String namespace,
+      Map<String, String> fragments,
+      Map<String, DartType> types,
+      List<ObjectType> enumObjectTypes,
+      List<ObjectType> interfaceObjectTypes,
+      List<ObjectType> unionObjectTypes,
+      List<ObjectType> objectObjectTypes,
+      List<ObjectType> inputObjectTypes,
+      ObjectType mutationObjectType) {
     var result = '';
     var emitter = DartEmitter(Allocator());
+    Map<String, Class> classes = {};
+
     List<String> enumString =
-    EnumGenerator().enumGenerator(enumTypes, namespace);
-    Map<String, Class> interfaces =
-    InterfaceGenerator().interfaceGenerator(interfaceTypes, namespace);
+        EnumGenerator().enumGenerator(enumObjectTypes, namespace: namespace);
+    Map<String, Class> interfaces = InterfaceGenerator().generate(
+        types,
+        enumObjectTypes,
+        interfaceObjectTypes,
+        unionObjectTypes,
+        objectObjectTypes,
+        inputObjectTypes,
+        namespace: namespace);
     Map<String, Class> unions =
-    UnionGenerator().unionGenerator(unionTypes, namespace);
-    Map<String, Class> inputs = InputObjectGenerator()
-        .inputObjectGenerator(inputObjectTypes, namespace, enumTypes);
-    Map<String, Class> objects = ObjectClassGenerator()
-        .objectClassGenerator(objectTypes, namespace, enumTypes, unionTypes);
+        UnionGenerator().generate(unionObjectTypes, namespace: namespace);
+    Map<String, Class> inputs = InputObjectGenerator().generate(
```

Comment:
What's the advantage for having any of these as a map? I think it's best to make it a `Iterable<Class>`. This would be simpler in my opinion and we would also have a more stable ordering in the generated code.

Reply:
We are using the `Map<String,Class>` to identify and get the Class object easily when we generate the Mutation object.
https://github.com/TurtleAI/graphql_generator/blob/75d5fb19bd6d82fe1b48c5f9791367b50c60f0d1/lib/generator/mutation.dart#L126-L138

Reply:
@justinjoyn I understand but if only the mutation class needs a map, maybe it's best for all the methods to return `Iterable<Class>`. But for the mutation logic, we convert it to a map. We shouldn't need to complicate all of the return type logic just for this one scenario.

You could create a helper method `Map<String, T> keyBy<T>(Iterable<T> list, String Function getKey(T item))`

What do you think? Would it simplify things a bit?

Reply:
Sounds right. Will try it.